tests/unit/test_factory.py

```diff
@@ -46,6 +46,42 @@ def test_create_fails_if_config_does_not_contain_type():
         Factory.create({"processorname": {"other": "value"}}, logger)
 
 
+@mark.parametrize("component", [None, {}])
+def test_create_fails_if_config_component_is_empty(component):
+    with raises(InvalidConfigurationError, match=r"The component definition is empty\."):
+        Factory.create(component, logger)
+
+
+@mark.parametrize("component", ["foo", 1, True, ["foo"], [], "", 0, False])
+def test_create_fails_if_config_component_is_no_dict(component):
+    with raises(
+        InvalidConfigurationError, match=r"The configuration must be specified as an object\."
+    ):
+        Factory.create(component, logger)
+
+
+def test_create_fails_if_config_sub_component_is_empty():
+    with raises(InvalidConfigurationError, match=r'The definition of component "foo" is empty\.'):
+        Factory.create({"foo": None}, logger)
+
+
+def test_create_fails_if_config_sub_component_has_no_type_specification():
+    with raises(
+        NoTypeSpecifiedError,
+        match=r"The type specification is missing for element with name 'foo'",
+    ):
+        Factory.create({"foo": {}}, logger)
+
+
+@mark.parametrize("component", ["foo", 1, True, ["foo"], [], "", 0, False])
+def test_create_fails_if_config_sub_component_is_no_dict(component):
+    with raises(
+        InvalidConfigurationError,
+        match=r'The configuration for component "foo" must be specified as an object\.',
+    ):
+        Factory.create({"foo": component}, logger)
+
+
```

Comment:
Thanks for implementing these test cases. :+1: I think you could simplify it a bit or merge them together though. All these tests expect an exception and are only calling the factory create method. If you consider the Error and the error-match-string as parameters as well, then I think this could be written in one parameterized test. You can have a look at the file `test_field_manager_rule.py`, there is a similar implementation.